rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/TechnologyUsageStatisticsService.java

```diff
@@ -0,0 +1,384 @@
+package org.jboss.windup.rules.apps.javaee;
+
+import java.io.Serializable;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+import java.util.stream.StreamSupport;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.jboss.windup.graph.GraphContext;
+import org.jboss.windup.graph.frames.TypeAwareFramedGraphQuery;
+import org.jboss.windup.graph.model.BelongsToProject;
+import org.jboss.windup.graph.model.ProjectModel;
+import org.jboss.windup.graph.model.WindupVertexFrame;
+import org.jboss.windup.graph.model.resource.FileModel;
+import org.jboss.windup.graph.service.GraphService;
+import org.jboss.windup.rules.apps.java.model.JavaClassModel;
+import org.jboss.windup.rules.apps.javaee.model.stats.GeneralStatsItemModel;
+import org.jboss.windup.rules.apps.javaee.model.stats.TechnologyUsageStatisticsModel;
+
+import com.tinkerpop.blueprints.Vertex;
+import com.tinkerpop.frames.FramedGraphQuery;
+import com.tinkerpop.pipes.filter.BackFilterPipe;
+import com.tinkerpop.pipes.transform.OutPipe;
+import com.tinkerpop.pipes.util.Pipeline;
+import com.tinkerpop.pipes.util.StartPipe;
+import org.jboss.windup.util.Logging;
+
+/**
+ * Provides CRUD methods for accessing the {@link TechnologyUsageStatisticsModel} vertices.
+ *
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ */
+public class TechnologyUsageStatisticsService extends GraphService<TechnologyUsageStatisticsModel>
+{
+    private static Logger LOG = Logging.get(TechnologyUsageStatisticsService.class);
+
+    /**
+     * Creates an instance of this service.
+     */
+    public TechnologyUsageStatisticsService(GraphContext context)
+    {
+        super(context, TechnologyUsageStatisticsModel.class);
+    }
+
+    public TechnologyUsageStatisticsModel get(ProjectModel projectModel, String technologyName)
+    {
+        Iterable<TechnologyUsageStatisticsModel> byName = findAllByProperty(TechnologyUsageStatisticsModel.NAME, technologyName);
+        TechnologyUsageStatisticsModel result = null;
+
+        for (TechnologyUsageStatisticsModel candidate : byName)
+        {
+            if (candidate.getProjectModel().equals(projectModel))
+            {
+                result = candidate;
+                break;
+            }
+        }
+
+        if (result == null)
+        {
+            result = create();
+            result.setComputed(new Date());
+            result.setProjectModel(projectModel);
+            result.setName(technologyName);
+            result.setOccurrenceCount(0);
```

Comment:
Why is default value `1` if you always set it to `0`?

Reply:
Because this is just to keep it from being null. There is no way to tell what the value should be from this point in the code and it doesn't really make sense to pass in an initial value, IMO.